dotCMS/src/main/java/com/dotcms/health/servlet/HealthProbeServlet.java

```diff
@@ -0,0 +1,684 @@
+    private volatile boolean lastReadinessHadDegraded = false; // Track if degraded conditions exist
+    private volatile long livenessCheckCount = 0;
+    private volatile long readinessCheckCount = 0;
+    
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        
+        // Initialize WebResource for authentication
+        webResource = new WebResource();
+        
+        // Use the centralized ObjectMapper provider for consistent JSON serialization
+        objectMapper = DotObjectMapperProvider.getInstance().getDefaultObjectMapper();
+        
+        // Get singleton health state manager instance (shared with CDI services)
+        healthStateManager = HealthStateManager.getInstance();
+        
+        // Register core health checks that work without CDI (for early liveness checks)
+        registerCoreHealthChecks();
+        
+        // Start background health monitoring
+        healthStateManager.initialize();
+        
+        // Try to set up CDI integration if available (non-blocking)
+        // This will discover and register CDI health checks like DatabaseHealthCheck, CacheHealthCheck, etc.
+        setupCdiIntegration();
+        
+        Logger.info(this, "HealthProbeServlet initialized with core health checks and CDI coordination");
+    }
+    
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) 
+            throws ServletException, IOException {
+        
+        // Get the request URI to determine which endpoint was called
+        String requestURI = request.getRequestURI();
+        String contextPath = request.getContextPath();
+        String servletPath = request.getServletPath();
+        
+        // Extract the actual endpoint from the request
+        String endpoint = servletPath;
+        
+        // Route to appropriate handler based on endpoint
+        switch (endpoint) {
+            case "/livez":
+                // K8s liveness probe - minimal text response, public access
+                handleLivenessProbe(request, response, true);
+                break;
+            case "/readyz":
+                // K8s readiness probe - minimal text response, public access  
+                handleReadinessProbe(request, response, true);
+                break;
+            case "/health/live":
+                // Detailed liveness JSON response, requires authentication
+                if (!requireAuthentication(request, response)) return;
+                handleLivenessProbe(request, response, false);
+                break;
+            case "/health/ready":
+                // Detailed readiness JSON response, requires authentication
+                if (!requireAuthentication(request, response)) return;
+                handleReadinessProbe(request, response, false);
+                break;
+            case "/health":
```

Comment:
I'm concerned that the `/health`  path now being reserved might break some customers (healthcare, beauty products) - it would mean that /health is no longer an allowed folder.

Reply:
hmm,  yeah you are correct.  I could just remove this and leave the /api/v1/health endpoint it does need to be under some path I was just trying to provide a separate infrastructure endpoint to our regular api,  Quarkus itself had issues like this and moved all their own infrastructure endpoints under a /q/* prefix so they could all be filtered by path in the infrastructure and not conflict.   The /readyz and /livez are a bit of a standard and not as normal to use and I don't think customers should be overriding these endpoints.   For the time being it is not important so just providing /api/v1/health, /readyz and /livez should be fine and we can change this later if we want to add.